llvm/lib/Target/SPIRV/SPIRVEmitIntrinsics.cpp

```diff
@@ -188,6 +188,40 @@ bool isConvergenceIntrinsic(const Instruction *I) {
          II->getIntrinsicID() == Intrinsic::experimental_convergence_loop ||
          II->getIntrinsicID() == Intrinsic::experimental_convergence_anchor;
 }
+
+bool isInternalNonVoidIntrinsic(const Value *I) {
+  if (const auto *II = dyn_cast<IntrinsicInst>(I))
+    switch (II->getIntrinsicID()) {
```

Comment:
Almost, but not exactly. This list doesn't include int_spv_gep, for example, because it returns ptr type. This list is to include internal intrinsics that returns a non-ptr value and so may potentially participate in fake_use emission.

Reply:
In that case, isn't the return type fetchable from the CallBase parent class? Or am I missing something? (Seems like maintaining a list like that is bound to end up wrong)

Reply:
I think you are right. My motivation was to explore the topic further and widen applicability of this PR's way to pass info through passes, but we may extend/address this later if needed. Another motivation was that internal intrinsics are intended for quite different things, and it's hard to separate them as a class to use in conditions, but this again doesn't matter in case of this PR specifically.

So for goals of this PR it may be better indeed to be more general and don't list intrinsics explicitly, and I've changed the explicit list to `F->getName().starts_with("llvm.spv.")` to exclude internal SPIR-V backend intrinsics.